opal/mca/threads/mutex.h

```diff
@@ -48,52 +51,127 @@ typedef struct opal_mutex_t opal_recursive_mutex_t;
 
 #include MCA_threads_mutex_base_include_HEADER
```

Comment:
I have a remark here related to the way we handle the modules selected during configure time. My remark raises from the fact that most editors have issues handling correctly header files defined via #define.

Can we replace our current approach with creating a link into the build tree, let's call it `selected`, to the selected module in the source tree to be used (this will work due to our include order, first build tree then source tree) ? Thus the code for all modules selected at configure time will look like this
```
#include "opal/mca/thread/selected/thread.h"
```
@jsquyres @bwbarrett

Reply:
@bosilca Thanks. I should be a good idea, but if allowed, may I create another PR for this? If you could post an issue, I/we/someone can work on it.

This PR neither introduced nor modified that header part. Even if this PR is for refactoring, I don't want to include everything in this single PR, which just adds a rebase/review cost, considering that this PR gets large.

Reply:
If we have such a discussion it can be outside the context of this PR.